plugins/CircuitExplorer/api/neuron/NeuronMorphology.cpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2015-2022, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Nadir Roman <nadir.romanguerrero@epfl.ch>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "NeuronMorphology.h"
+
+NeuronMorphology::NeuronMorphology(std::vector<Section> sections, std::optional<Soma> soma)
+    : _sections(std::move(sections))
+    , _soma(std::move(soma))
+{
+}
+
+bool NeuronMorphology::hasSoma() const noexcept
+{
+    return _soma.has_value();
```

Comment:
Can return soma directly.

Reply:
Cannot be changed to return soma, there is no implicit call to the bool() operator